python/pyarrow/_dataset.pyx

```diff
@@ -414,10 +419,48 @@ cdef class FileFragment(Fragment):
     @property
     def path(self):
         """
-        The path of the data file viewed by this fragment.
+        The path of the data file viewed by this fragment, if it views a
+        file. If instead it views a buffer, this will be "<Buffer>".
         """
         return frombytes(self.file_fragment.source().path())
 
+    def open(self):
```

Comment:
Ideally I'd just provide a `filesystem` property for this fragment. Unfortunately at this stage we don't have a `shared_ptr[CFileSystem]` so we can't wrap it as a `FileSystem`